src/roomspace.c

```diff
@@ -834,29 +1049,134 @@ static void keeper_update_roomspace(struct RoomSpace *roomspace)
             roomspace->is_active = false;
             return;
         }
-        keeper_build_room(slab_subtile(roomspace->buildx, 0), slab_subtile(roomspace->buildy, 0),
-                          roomspace->plyr_idx, roomspace->rkind);
+            keeper_build_room(slab_subtile(roomspace->buildx, 0), slab_subtile(roomspace->buildy, 0),
+            roomspace->plyr_idx, roomspace->rkind);
     }
     // find next point
-    do
+    if (roomspace->drag_mode)
     {
-        roomspace->buildx++;
-        if (roomspace->buildx > roomspace->right)
+        switch (roomspace->drag_direction)
         {
-            roomspace->buildx = roomspace->left;
-            roomspace->buildy++;
-        }
-        if (!roomspace->is_roomspace_a_box)
-        {
-            find_next_point(roomspace);
+            case 0:
```

Comment:
Drag directions 0~3 are not informative. Could you use an enum or comment the directions?